### DISCUSSION TITLE
How to debug revalidation?

### DISCUSSION THREAD

**Discussion by good-idea** [vercel/next.js#17634] [Help]
I'm having an issue where my static props aren't updating after the specified `revalidate` setting. (In my case it's `60`). It's difficult to debug since it's only happening in production, since the dev mode will always refetch. I don't see any errors or issues in my API requests in the vercel dashboard.

Is there some way I can better dive into what's going on? Here is a sample page:

```
import * as React from 'react'
import { GetStaticProps } from 'next'
import { getDataFromTree } from '@apollo/client/react/ssr'
import { ssrClient, App, ShopView } from '../../src/views'

const Shop = () => {
  return <ShopView />
}

export const getStaticProps: GetStaticProps = async () => {
  const StaticApp = (
    <App>
      <ShopView />
    </App>
  )

  await getDataFromTree(StaticApp)
  const apolloCache = ssrClient.extract()

  return { props: { apolloCache }, revalidate: 60 }
}

export default Shop
```

It *may* be a caching issue with apollo. But I'm not sure how to tell when `getStaticProps` is running to re-validate the page data.

**Comment by alex-r89** [comment]
I seem to be having a similar issue. Did you find a solution?

**Reply by good-idea** [reply]
I wasn't able to find a way to debug anything, but I changed the way I am handling apollo by forcing it to clear its cache before each request in getStaticProps:

https://github.com/good-idea/hikawa.studio/blob/main/app/pages/shop/index.tsx

https://github.com/good-idea/hikawa.studio/blob/main/app/src/utils/ssr.tsx#L54

**Reply by alex-r89** [reply]
I thought my issue was also related to this, but I found out my backend graphQL query was erroring and timing out. I noticed in the Function Logs on the Vercel dashboard that my query was timing out after 10 seconds which was causing the props never to get passed, because the never updated.

So for me it actually wasn't related to revalidate.
